pkg/cmd/deploy/worker.go

```diff
@@ -11,44 +11,50 @@ import (
 )
 
 // worker reads the range of jobs and uploads the file, if there is an error during upload, we returning it through the results channel
-func worker(jobs <-chan contracts.FileOps, results chan<- error, currentFile *int64, clientUpload *storage.Client, conf *contracts.AzionApplicationOptions) {
+func worker(jobs <-chan contracts.FileOps, results chan<- error, currentFile *int64, client *storage.Client, conf *contracts.AzionApplicationOptions) {
+	var lastError error
 	for job := range jobs {
-		// Once ENG-27343 is completed, we might be able to remove this piece of code
-		fileInfo, err := job.FileContent.Stat()
-		if err != nil {
-			logger.Debug("Error while worker tried to read file stats", zap.Error(err))
-			results <- err
-			return
-		}
+		for Retries <= 5 {
+			fileInfo, err := job.FileContent.Stat()
+			if err != nil {
+				logger.Debug("Error while worker tried to read file stats", zap.Error(err))
+				lastError = err
+				break
+			}
 
-		// Check if the file size is zero
-		if fileInfo.Size() == 0 {
-			logger.Debug("\nSkipping upload of empty file: " + job.Path)
-			results <- nil
-			atomic.AddInt64(currentFile, 1)
-			return
-		}
+			if fileInfo.Size() == 0 {
+				logger.Debug("skipping upload of empty file", zap.String("path", job.Path))
+				results <- nil
+				atomic.AddInt64(currentFile, 1)
+				break
+			}
 
-		if err := clientUpload.Upload(context.Background(), &job, conf); err != nil {
-			logger.Debug("Error while worker tried to upload file: <"+job.Path+"> to storage api", zap.Error(err))
-			for Retries < 5 {
-				atomic.AddInt64(&Retries, 1)
-				logger.Debug("Retrying to upload the following file: <"+job.Path+"> to storage api", zap.Error(err))
-				err := clientUpload.Upload(context.Background(), &job, conf)
+			if err := client.Upload(context.Background(), &job, conf); err != nil {
+				logger.Debug("Error while worker tried to upload file: <"+job.Path+"> to storage api", zap.Error(err))
+				Retries += 1
+				_, err := job.FileContent.Seek(0, 0)
 				if err != nil {
+					logger.Debug("An error occurred while seeking fileContent", zap.Error(err))
+					break
+				}
+
+				if Retries < 5 {
+					logger.Debug("Retrying to upload the following file: <"+job.Path+"> to storage api", zap.Error(err))
 					continue
+				} else {
+					break
 				}
-				break
 			}
 
-			if Retries >= 5 {
-				logger.Debug("There have been 5 retries already, quitting upload")
-				results <- err
-				return
-			}
+			atomic.AddInt64(currentFile, 1)
+			results <- nil
+			break
 		}
 
-		atomic.AddInt64(currentFile, 1)
-		results <- nil
+		if Retries > 5 || lastError != nil {
+			logger.Debug("There have been 5 retries already, quitting upload")
```

Comment:
If there is an error when reading the file, the debug message will say 5 retries, even though this did not happen 🤔 
am I correct in assuming this?